src/main/java/com/arrl/radiocraft/common/network/serverbound/SHandheldRadioUpdatePacket.java

```diff
@@ -19,20 +19,27 @@ public class SHandheldRadioUpdatePacket implements CustomPacketPayload {
     int index;
     boolean powered;
     boolean ptt;
+    float gain;
+    float micGain;
     int frequencyKiloHertz; //TODO change to allow server to control frequency steps
 
-    public SHandheldRadioUpdatePacket(int index, boolean powered, boolean ptt, int frequencyKiloHertz) {
+    public SHandheldRadioUpdatePacket(int index, boolean powered, boolean ptt, float gain, float micGain, int frequencyKiloHertz) {
         this.index = index;
         this.powered = powered;
         this.ptt = ptt;
+        this.gain = gain;
+        this.micGain = micGain;
         this.frequencyKiloHertz = frequencyKiloHertz;
     }
 
     //constructor for StreamCodec decoding
-    private SHandheldRadioUpdatePacket(Integer index, Byte packed, Integer frequencyKiloHertz) {
+    private SHandheldRadioUpdatePacket(Integer index, Byte packed, Byte gain, Byte micGain, Integer frequencyKiloHertz) {
         this.index = index;
         this.powered = (packed&0x1)==0x1;
         this.ptt = (packed&0x2)==0x2;
+        // byte & 0xFF here will implicitly cast byte to int and get rid of sign, restoring 0-255 range
+        this.gain = (float)((gain & 0xFF) / 10.0f);
+        this.micGain = (float)((micGain & 0xFF) / 10.0f);
```

Comment:
I'm not sure why we bother converting to/from a byte and float here. Can we just preserve the floats?